rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake

```diff
@@ -18,37 +18,33 @@ set(_generated_msg_header_files "")
 set(_generated_msg_source_files "")
 set(_generated_srv_header_files "")
 set(_generated_srv_source_files "")
+set(_generated_action_header_files "")
+set(_generated_action_source_files "")
 foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
   get_filename_component(_extension "${_idl_file}" EXT)
   get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
   get_filename_component(_parent_folder "${_parent_folder}" NAME)
   get_filename_component(_msg_name "${_idl_file}" NAME_WE)
   string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
-
   if(_extension STREQUAL ".msg")
-    if(_parent_folder STREQUAL "msg")
-      list(APPEND _generated_msg_header_files
-        "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_introspection_c.h")
-      list(APPEND _generated_msg_source_files
-        "${_output_path}/${_parent_folder}/${_header_name}__type_support.c")
-    elseif(_parent_folder STREQUAL "srv")
-      list(APPEND _generated_srv_header_files
-        "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_introspection_c.h")
-      list(APPEND _generated_srv_source_files
-        "${_output_path}/${_parent_folder}/${_header_name}__type_support.c")
-    else()
+    set(_allowed_parent_folders "msg" "srv" "action")
+    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
       message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
     endif()
   elseif(_extension STREQUAL ".srv")
-    list(APPEND _generated_srv_header_files
-      "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_introspection_c.h"
-    )
-    list(APPEND _generated_srv_source_files
-      "${_output_path}/${_parent_folder}/${_header_name}__type_support.c"
-    )
+    set(_allowed_parent_folders "srv" "action")
+    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
+      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
+    endif()
   else()
     message(FATAL_ERROR "Interface file with unknown extension: ${_idl_file}")
   endif()
+  set(_generated_source_files "_generated_${_parent_folder}_source_files")
+  set(_generated_header_files "_generated_${_parent_folder}_header_files")
```

Comment:
`_generated_header_files` doesn't appear to be used anywhere. Possible cause of this CMake Error output?

```
CMake Error at /home/developer/workspaces/ros2-dev0/install/rosidl_typesupport_introspection_c/share/rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake:113 (add_library):
  Cannot find source file:

    /home/developer/workspaces/ros2-dev0/build/test_msgs/rosidl_typesupport_introspection_c/test_msgs/action/shane_dynamic_array_primitives_nested__rosidl_typesupport_introspection_c.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx
Call Stack (most recent call first):
  /home/developer/workspaces/ros2-dev0/install/ament_cmake_core/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:38 (include)
  /home/developer/workspaces/ros2-dev0/install/rosidl_cmake/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:186 (ament_execute_extensions)
  CMakeLists.txt:42 (rosidl_generate_interfaces)


CMake Error: CMake can not determine linker language for target: test_msgs__rosidl_typesupport_introspection_c
CMake Error: Cannot determine link language for target "test_msgs__rosidl_typesupport_introspection_c".
```

Reply:
Ah I misread the code, looks like the issue is `_generated_action_source_files` is missing from a call to `add_custom_comand`

Reply:
Yeah, that seems to be the case.